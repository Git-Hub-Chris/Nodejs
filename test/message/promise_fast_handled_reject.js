'use strict';

const common = require('../common');
const assert = require('assert');

const p1 = new Promise((res, rej) => {
  throw new Error('One');
});

new Promise((res, rej) => {
  throw new Error('Two');
});

const p3 = new Promise((res, rej) => {
  throw new Error('Three');
});

process.nextTick(() => {
  setTimeout(common.mustCall(() => {
    p1.catch(() => {});
    p3.catch(() => {});
  }), 1);
});

process.on('rejectionHandled', () => {}); // Ignore

process.on('uncaughtException', (err) =>
  assert.fail('Should not trigger uncaught exception'));

process.on('exit', () => process._rawDebug('exit event emitted'));
