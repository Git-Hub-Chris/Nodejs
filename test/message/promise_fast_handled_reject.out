(node:*) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): One
(node:*) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): Two
(node:*) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 3): Three
exit event emitted
*test*message*promise_fast_handled_reject.js:*
  throw new Error('Two');
  ^

Error: Two
    at *test*message*promise_fast_handled_reject.js:*:*
    at Promise (<anonymous>)
    at Object.<anonymous> (*test*message*promise_fast_handled_reject.js:*:*)
    at Module._compile (module.js:*:*)
    at Object.Module._extensions..js (module.js:*:*)
    at Module.load (module.js:*:*)
    at tryModuleLoad (module.js:*:*)
    at Function.Module._load (module.js:*:*)
    at Function.Module.runMain (module.js:*:*)
    at startup (bootstrap_node.js:*:*)
